jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecDependencyActor.scala

```diff
@@ -0,0 +1,134 @@
+package dcos.metronome
+package jobspec.impl
+
+import java.time.Instant
+
+import akka.actor.{Actor, ActorLogging, Props, Stash, Status}
+import akka.pattern.pipe
+import com.typesafe.scalalogging.StrictLogging
+import dcos.metronome.jobrun.{JobRunService, StartedJobRun}
+import dcos.metronome.jobspec.impl.JobSpecDependencyActor.{DependenciesState, UpdateJobSpec}
+import dcos.metronome.model.{Event, JobId, JobRun, JobRunStatus, JobSpec}
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+
+/**
+  * Manages one JobSpec.
+  *
+  * If the JobSpec has dependencies, it subscribes to the events of its dependencies, ie parents.
+  *
+  * This actor is analog to [[JobSpecSchedulerActor]].
+  *
+  * TODO: find a better name
+  */
+class JobSpecDependencyActor(initSpec: JobSpec, runService: JobRunService) extends Actor with Stash with ActorLogging {
+
+  private[impl] var spec = initSpec
+  private[impl] var lastSuccessfulRun: Instant = Instant.MIN
+  val dependenciesState = DependenciesState(initSpec.dependencies.toSet)
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  override def preStart(): Unit = {
+    super.preStart()
+    context.system.eventStream.subscribe(self, classOf[Event.JobRunEvent])
+  }
+
+  override def postStop(): Unit = {
+    context.system.eventStream.unsubscribe(self)
+    super.postStop()
+  }
+
+  override def receive: Receive = idling
+
+  def idling: Receive = {
+    case ev: Event.JobRunEvent =>
+      dependenciesState.update(ev.jobRun)
+      if (dependenciesState.shouldTriggerJob(lastSuccessfulRun)) {
+        log.info(s"Start next run of job ${spec.id}, all parents finished successfully")
+        runService.startJobRun(initSpec).pipeTo(self)
+        context.become(running)
```

Comment:
I told @pierrebeitz already what limitation we have with this.

First, I think we need to handle the result of `startJobRun`. It stands to question whether we need another receive method for that.

Second and this is more important, I saw job runs triggered each time there was an update. Imagine C run five seconds depends on A and B. B is run every ten seconds but A every ten millis. This would trigger C multiple times.

A way to avoid this is to save the `lastCreatedAt` for C and use this as well to check whether we already scheduled a run for C or not. I will create a follow up as a proposal.